workers/unity/Packages/com.improbable.gdk.transformsynchronization/Systems/DefaultApplyLatestTransformSystem.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Generated.Improbable.Transform;
+using Improbable.Gdk.Core;
+using Unity.Collections;
+using Unity.Entities;
+using UnityEngine;
+using UnityEngine.Experimental.PlayerLoop;
+
+namespace Improbable.Gdk.TransformSynchronization
+{
+    [DisableAutoCreation]
+    [UpdateInGroup(typeof(FixedUpdate))]
+    public class DefaultApplyLatestTransformSystem : ComponentSystem
+    {
+        private struct RigidbodyData
+        {
+            [ReadOnly] public readonly int Length;
+            [ReadOnly] public ComponentDataArray<TransformToSet> CurrentTransform;
+            public ComponentArray<Rigidbody> Rigidbody;
+
+            [ReadOnly] public ComponentDataArray<NotAuthoritative<TransformInternal.Component>>
+                DenotesNotAuthoritative;
+        }
+
+        private struct TransformData
+        {
+            [ReadOnly] public readonly int Length;
+            [ReadOnly] public ComponentDataArray<TransformToSet> CurrentTransform;
+            [ReadOnly] public ComponentArray<Transform> Transform;
```

Comment:
No readonly required, `ComponentArray` only has a Get

Reply:
It's just there to make it clear that we are not writing to the transform here. Can remove it if we think this is going to be an error in future but right now it gives information to the reader.